Blockzilla/OverlayView.swift

```diff
@@ -104,29 +104,33 @@ class OverlayView: UIView {
     func setSearchQuery(query: String, animated: Bool) {
         searchQuery = query
         let query = query.trimmingCharacters(in: .whitespaces)
-        
-        if false, let pasteBoard = UIPasteboard.general.string {
-            if let pasteURL = URL(string: pasteBoard), pasteURL.isWebPage() {
-                let attributedURL = NSAttributedString(string: pasteURL.absoluteString, attributes: [NSAttributedStringKey.foregroundColor: UIColor.white])
-                    copyButton.setAttributedTitle(attributedURL, for: .normal)
-            }
-            else {
-                setAttributedButtonTitle(phrase: pasteBoard, button: copyButton)
+
+        var showCopyButton = false
+
+        UIPasteboard.general.urlAsync() {
+            let handoffUrl = $0
+            DispatchQueue.main.async {
+                if let url = handoffUrl, url.isWebPage() {
+                    self.copyButton.setAttributedTitle(NSAttributedString(string: String(format: UIConstants.strings.linkYouCopied, url.absoluteString), attributes: [NSAttributedStringKey.foregroundColor: UIColor.white]), for: .normal)
+                    showCopyButton = url.isWebPage()
+                }
+
+                // Show or hide the search button depending on whether there's entered text.
+                if self.searchButton.isHidden != query.isEmpty {
+                    let duration = animated ? UIConstants.layout.searchButtonAnimationDuration : 0
+                    self.searchButton.animateHidden(query.isEmpty, duration: duration)
+                    self.searchBorder.animateHidden(query.isEmpty, duration: duration)
+                }
+                self.setAttributedButtonTitle(phrase: query, button: self.searchButton)
+                self.updateCopyConstraint(showCopyButton: showCopyButton)
             }
+            self.setAttributedButtonTitle(phrase: query, button: self.searchButton)
+            self.updateCopyConstraint(showCopyButton: showCopyButton)
```

Comment:
Did you want this outside the `urlAsync` call? I'm getting a crasher here because we're updating UI on a background thread.